### PR TITLE
python312Packages.nexusformat: init at 1.0.6

### PR DESCRIPTION
**Pull request by oberth-effect**
## Description of changes
Add `nexusformat` Python library.

https://pypi.org/project/nexusformat/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oberth-effect** [comment]
Added tests.

Output of `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 336081"` 
```
2 packages added:
python311Packages.nexusformat (init at 1.0.6) python312Packages.nexusformat (init at 1.0.6)

...

4 packages built:
python311Packages.nexusformat python311Packages.nexusformat.dist python312Packages.nexusformat python312Packages.nexusformat.dist
```

File: pkgs/development/python-modules/nexusformat/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    inherit pname version;
+    hash = "sha256-UxU3PA/2r/uamdysbfC0L2JinHgfkXhssHIo2hf3zlA=";
+  };
+
+  pyproject = true;
+
+  build-system = [
+    setuptools
+    setuptools-scm
+    wheel
```
Comment (natsukium) [inline pkgs/development/python-modules/nexusformat/default.nix:34]:
```suggestion
```
`wheel` is not a `build-system` dependencies.

File: pkgs/development/python-modules/nexusformat/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    hdf5plugin
+    numpy
+    scipy
+  ];
+
+  nativeCheckInputs = [ pytest ];
+
+  checkPhase = ''
+    pytest
+  '';
```
Comment (natsukium) [inline pkgs/development/python-modules/nexusformat/default.nix:48]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```
Could you please use `pytestCheckHook` instead?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

File: pkgs/development/python-modules/nexusformat/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    wheel
+  ];
+
+  dependencies = [
+    h5py
+    hdf5plugin
+    numpy
+    scipy
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/nexusformat/default.nix:43]: Could you enable `pythonImportsCheck` for a simple check?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

File: pkgs/development/python-modules/nexusformat/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  nativeCheckInputs = [ pytest ];
+
+  checkPhase = ''
+    pytest
+  '';
+
+  meta = with lib; {
+    description = "Python API to open, create, and manipulate NeXus data written in the HDF5 format.";
```
Comment (natsukium) [inline pkgs/development/python-modules/nexusformat/default.nix:51]:
```suggestion
    description = "Python API to open, create, and manipulate NeXus data written in the HDF5 format";
```
Please remove the period.
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 336081` at c9da362b run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python311Packages.nexusformat</li>
    <li>python312Packages.nexusformat</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

Welcome to the nixpkgs!🎉

**Pull request merged**